projects/solar/inc/solar_fsm.h

```diff
@@ -0,0 +1,31 @@
+#pragma once
+
+// FSM for managing the relay state.
+// Requires the event queue, GPIO, and the DRV120 relay driver to be initialized.
+
+#include <stdbool.h>
+#include <stdint.h>
+
+#include "event_queue.h"
+#include "fsm.h"
+#include "status.h"
+
+#define MAX_RELAY_OPEN_EVENTS 8
+
+typedef struct SolarFsmSettings {
+  // Fault events which cause the relay to open.
+  EventId relay_open_events[MAX_RELAY_OPEN_EVENTS];
+  uint8_t num_relay_open_events;  // length of preceding array
+} SolarFsmSettings;
```

Comment:
Do we ever expect the number of relay open events to change?

Reply:
I think it could change if hardware thinks of or finds more faults that require isolation. It shouldn't change during runtime though, certainly.